app/controllers/v0/terms_of_use_agreements_controller.rb

```diff
@@ -31,6 +33,10 @@ def decline
       render_error(e.message)
     end
 
+    rescue_from TermsOfUse::Exceptions::CommonNameNotFoundError do |error|
+      render_error(error.to_s)
```

Comment:
As written, this returns this as the error:

```json
{
    "error": "TermsOfUse::Exceptions::CommonNameNotFoundError"
}
```

Did we want to respond with something different for the error message? I think the class name is a bit opaque, maybe something like `{ error: "Invalid common name for user." }`? You could refactor the exception class to something like this:

```ruby
module TermsOfUse
  module Exceptions
    class CommonNameNotFoundError < StandardError
      attr_reader :common_name

      def initialize(common_name:)
        super()
        @common_name = common_name
      end

      def message
        "Invalid common name provided for user; common_name: #{common_name}."
      end
    end
  end
end
```

...and then update the service object:

```ruby
    def validate_common_name
      raise Exceptions::CommonNameNotFoundError.new(common_name:) if common_name.blank?
    end
```

...and then the controller:

```ruby
    rescue_from TermsOfUse::Exceptions::CommonNameNotFoundError do |error|
      render_error(error.message)
    end
```

Reply:
oh I see Dick had the same feedback here (around rendering the message not the error itself)

Reply:
I think a good way forward would be a union between our suggestions. I think you are right; a `CommonNameMissing` error seems more appropriate, in which case we can refactor the exception class to something like:

```ruby
module TermsOfUse
  module Exceptions
    class CommonNameMissingError < StandardError
      def message
        "Name for user must be present"
      end
    end
  end
end
```

...or just send the message when we raise the error:

```ruby
raise Exceptions::CommonNameMissingError, 'Name for user must be present' if common_name.blank?
```